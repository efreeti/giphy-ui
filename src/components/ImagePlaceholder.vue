<template>
  <div class="image-placeholder">
    <span class="image-placeholder-content">
      <slot />
    </span>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class ImagePlaceholder extends Vue {
}
</script>

<style lang="scss" scoped>
.image-placeholder {
  width: 400px;
  height: 400px;
  line-height: 400px;
}

.image-placeholder-content {
  display: inline-block;
  vertical-align: middle;
  line-height: normal;
}
</style>
