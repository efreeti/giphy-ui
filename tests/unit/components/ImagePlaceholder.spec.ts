import { mount } from '@vue/test-utils';

import '../config';

import ImagePlaceholder from '@/components/ImagePlaceholder.vue';

describe('ImagePlaceholder', () => {
  it('should render correctly', () => {
    expect(mount(ImagePlaceholder, {
      slots: {
        default: '<span>some content</span>',
      },
    }))
        .toMatchSnapshot();
  });
});
