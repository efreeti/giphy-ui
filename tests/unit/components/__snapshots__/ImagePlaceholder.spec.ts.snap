// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePlaceholder should render correctly 1`] = `<div class="image-placeholder"><span class="image-placeholder-content"><span>some content</span></span></div>`;
